core/block_validator.go

```diff
@@ -94,9 +94,21 @@ func (v *BlockValidator) ValidateBody(block *types.WorkObject) error {
 		if hash := types.CalcUncleHash(block.Uncles()); hash != header.UncleHash() {
 			return fmt.Errorf("uncle root hash mismatch: have %x, want %x", hash, header.UncleHash())
 		}
-		if v.hc.ProcessingState() {
-			if hash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil)); hash != header.TxHash() {
-				return fmt.Errorf("transaction root hash mismatch: have %x, want %x", hash, header.TxHash())
+		if hash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil)); hash != header.TxHash() {
+			return fmt.Errorf("transaction root hash mismatch: have %x, want %x", hash, header.TxHash())
+		}
+		activeLocations := common.NewChainsAdded(v.hc.currentExpansionNumber)
+		for _, tx := range block.Transactions() {
+			for _, txo := range tx.TxOut() {
```

Comment:
This will panic for Quai or External transactions

Reply:
Ah, good catch

Reply:
fixed